### PR TITLE
Allow deletion of bounced, declined and cancelled referees

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

```
As a...candidate
I need to..delete previous references requests
So that...so that I feel in control of what’s shown
```

The summary card on the decoupled references review page needs delete request link for the cancelled, declined and bounced references.

## Changes proposed in this pull request

cancelled

|Before|After|
|------------|------------|
|image|image|

declined

|Before|After|
|------------|------------|
|image|image|

failed

|Before|After|
|------------|------------|
|image|image|

Confirm delete request page

image

## Link to Trello card

https://trello.com/c/PlcJ384z/2242-%F0%9F%92%94-dev-delete-a-cancelled-failed-declined-request

## Guidance for review

There is a subtle difference between the behaviour for this and the delete actions for the feedback_provided and not_requested_yet states. The provided and not requested states show 'Delete referee' link that goes to the  confirm_destroy view. These states show a  'Delete request' link and show the confirm_delete_request view.

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by paulrobertlloyd** [comment]
@davidgisbey Button says ‘Yes I’m sure - delete this reference_request’ (underscore in `reference_request`)

Screenshot 2020-10-22 at 11 26 03

**Comment by davidgisbey** [comment]
Good spot. Have fixed.

File: app/views/candidate_interface/decoupled_references/review/confirm_delete_request.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<% content_for :before_content, govuk_back_link_to %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: @reference, url: candidate_interface_destroy_decoupled_reference_path(@reference), method: :delete do |f| %>
+      <h1 class="govuk-heading-xl">
+        <%= t('page_titles.references_delete_request') %>
+      </h1>
+
+      <%= f.submit 'Yes I’m sure - delete this reference request' , class: 'govuk-button govuk-button--warning', data: { module: 'govuk-button' } %>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/decoupled_references/review/confirm_delete_request.html.erb:11]: Should button labels be in the localisation file?

File: app/views/candidate_interface/decoupled_references/review/confirm_delete_request.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+      <p class="govuk-body">
+        <%= govuk_link_to 'No, I’ve changed my mind', candidate_interface_decoupled_references_review_path %>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/decoupled_references/review/confirm_delete_request.html.erb:14]: Ditto, should this be localised?

**Review by paulrobertlloyd — APPROVED** [review]

File: app/views/candidate_interface/decoupled_references/review/confirm_delete_request.html.erb (hunk 1 above)
Comment (davidgisbey) [reply app/views/candidate_interface/decoupled_references/review/confirm_delete_request.html.erb]: Have done 👍

File: app/views/candidate_interface/decoupled_references/review/confirm_delete_request.html.erb (hunk 2 above)
Comment (davidgisbey) [reply app/views/candidate_interface/decoupled_references/review/confirm_delete_request.html.erb]: ditto

**Pull request merged**